pyproject.toml

```diff
@@ -69,6 +69,7 @@ dependencies = [
     "scikit-learn>=1.3",
     "scipy>=1.11",
     "Shapely>=2",
+    "splink>=3.9.11",
```

Comment:
It looks like [splink put out a new 3.9.12 release](https://github.com/moj-analytical-services/splink/releases/tag/v3.9.12) with full SQLAlchemy 2.x support (not just a packaging change), which might be good to use as the minimum allowed version.